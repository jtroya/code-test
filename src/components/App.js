import React, { Component } from 'react';

import Tutorial from './Tutorial';

class App extends Component {
  componentDidMount () {
    this.props.loadApp();
  }

  handleOpenModal = () => {
    this.props.openModal();
  }

  handleCloseModal = () => {
    this.props.skipTutorial();
  }
  
  render() {
    const { showTutorialModal } = this.props;
    return (
      <div className="App">
        <div className="body">
          <div className="container">
          <button className="btn-tutorial" onClick={ this.handleOpenModal }>Open</button>
          <Tutorial isOpen={ showTutorialModal } closeModal={ this.handleCloseModal } />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
