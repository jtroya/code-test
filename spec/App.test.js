import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '../src/components/App';

describe('App', () => {
  let wrapper;
  const loadApp = jest.fn();

  it('should render de App', () => {
    wrapper = shallow(<App loadApp={ loadApp } />);
    expect(wrapper.find('.App')).toBeTruthy();
    expect(wrapper.find('.body')).toBeTruthy();
    expect(wrapper.find('.container')).toBeTruthy();
    expect(wrapper.find('.btn-tutorial')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = shallow(<App loadApp={ loadApp } />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
