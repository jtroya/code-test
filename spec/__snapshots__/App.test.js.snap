// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="body"
  >
    <div
      className="container"
    >
      <button
        className="btn-tutorial"
        onClick={[Function]}
      >
        Open
      </button>
      <Tutorial
        closeModal={[Function]}
      />
    </div>
  </div>
</div>
`;
